app/controllers/lists/entities_controller.rb

```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module Lists
+  class EntitiesController < ApplicationController
+    before_action :authenticate_user!
+    before_action :block_restricted_user_access
+    before_action -> { current_user.raise_unless_can_edit! }
+    before_action :set_list, :set_entity
+
+    def create
+      if @entity.save
+        # Ought to be able to do this stuff with accepts_nested_attributes_for
```

Comment:
Agreed! Plenty of refactoring we could work on in the list and entity controllers too.